dpbench/benchmarks/gpairs/gpairs_sycl_native_ext/gpairs_sycl/_gpairs_sycl.cpp

```diff
@@ -0,0 +1,77 @@
+//==- _gpairs_sycl.cpp - Python native extension of Black-Scholes   ===//
+//
+// Copyright 2022 Intel Corp.
+//
+// SPDX - License - Identifier : Apache 2.0
+///
+/// \file
+/// The files implements a SYCL-based Python native extension for the
+/// gpairs benchmark.
+
+#include "_gpairs_kernel.hpp"
+#include <dpctl4pybind11.hpp>
+
+template <typename... Args> bool ensure_compatibility(const Args &...args)
+{
+    std::vector<dpctl::tensor::usm_ndarray> arrays = {args...};
+
+    auto arr = arrays.at(0);
+    auto q = arr.get_queue();
+    auto type_flag = arr.get_typenum();
+    auto arr_size = arr.get_size();
+
+    for (auto &arr : arrays) {
+        if (!(arr.get_flags() & (USM_ARRAY_C_CONTIGUOUS))) {
+            std::cerr << "All arrays need to be C contiguous.\n";
+            return false;
+        }
+        if (arr.get_typenum() != type_flag) {
+            std::cerr << "All arrays should be of same elemental type.\n";
+            return false;
+        }
+        if (arr.get_ndim() > 1) {
+            std::cerr << "All arrays expected to be single-dimensional.\n";
+            return false;
+        }
+    }
+    return true;
+}
+
+void gpairs_sync(size_t nopt,
+                 size_t nbins,
+                 dpctl::tensor::usm_ndarray x1,
+                 dpctl::tensor::usm_ndarray y1,
+                 dpctl::tensor::usm_ndarray z1,
+                 dpctl::tensor::usm_ndarray w1,
+                 dpctl::tensor::usm_ndarray x2,
+                 dpctl::tensor::usm_ndarray y2,
+                 dpctl::tensor::usm_ndarray z2,
+                 dpctl::tensor::usm_ndarray w2,
+                 dpctl::tensor::usm_ndarray rbins,
+                 dpctl::tensor::usm_ndarray results)
+{
+    if (!ensure_compatibility(x1, y1, z1, w1, x2, y2, z2, w2, rbins, results))
+        throw std::runtime_error("Input arrays are not acceptable.");
+
+    if (x1.get_typenum() != UAR_FLOAT) {
```

Comment:
How about allowing both `UAR_DOUBLE` and `UAR_FLOAT`? That way only driver code from Python will have to change.

The error message is not correct.